pkg/test/framework/components/galley/snapshot.go

```diff
@@ -19,7 +19,7 @@ import (
 	"fmt"
 	"reflect"
 
-	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/proto"
```

Comment:
Same here. Galley should only use gogo proto.

Reply:
gogo seems to be broken here:
```
panic: merger not found for type:intstr.Type [recovered]
 panic: merger not found for type:intstr.Type

goroutine 60 [running]:
testing.tRunner.func1(0xc0003dcc00)
 /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x10b4120, 0xc0003a9870)
 /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo(0xc000425540)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:662 +0xfb3
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc000425540, 0xc0003e92e8, 0xc0003ca668)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:113 +0x4a2
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func28(0xc0003e92e8, 0xc0003ca668)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:555 +0x3e
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc000425400, 0xc0003e92c0, 0xc0003ca640)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:139 +0x46a
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func27(0xc000452680, 0xc00042a000)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:545 +0x381
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc000424f00, 0xc000452680, 0xc00042a000)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:139 +0x46a
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Merge(0x1e35f20, 0x1465e60, 0xc000452680, 0x1465e60, 0xc00042a000)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:50 +0x4c
k8s.io/api/core/v1.(*ServiceSpec).XXX_Merge(0xc000452680, 0x1465e60, 0xc00042a000)
 /usr/local/google/home/howardjohn/go/pkg/mod/k8s.io/api@v0.0.0-20191003000013-35e20aa79eb8/core/v1/generated.pb.go:5165 +0x57
github.com/gogo/protobuf/proto.Merge(0x1465e60, 0xc000452680, 0x1465e60, 0xc00042a000)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/clone.go:95 +0x327
github.com/gogo/protobuf/proto.Clone(0x1465e60, 0xc00042a000, 0xc0003b6f00, 0x145e060)
 /usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/clone.go:52 +0x18f
istio.io/istio/galley/pkg/config/resource.(*Entry).Clone(0xc0003aa4d0, 0xc0003b6f00)
 /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/resource/entry.go:38 +0x134
istio.io/istio/galley/pkg/config/processor/transforms/serviceentry_test.TestAddOrder.func1.1(0xc0003dcc00, 0xc0003b6f00, 0xc0003b6f20, 0xc0003a69c0, 0x12a94f0, 0x7, 0x1, 0x12ba30b, 0x14, 0xc0003aa4d0, ...)
 /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/processor/transforms/serviceentry/create_test.go:482 +0x107
istio.io/istio/galley/pkg/config/processor/transforms/serviceentry_test.(*pipeline).run.func1(0xc0003dcc00)
 /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/processor/transforms/serviceentry/create_test.go:1167 +0x19c
testing.tRunner(0xc0003dcc00, 0xc0003ed980)
 /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
 /usr/local/go/src/testing/testing.go:960 +0x350
```

Wouldn't some tests fail if switching was a problem?

Reply:
No, it's a silent error. You eventually get some registration errors at runtime.

Reply:
See https://github.com/kubernetes/kubernetes/issues/83912. It can't be a completely silent error, right? I have deployed galley with these changes and see no problems, all e2e tests pass, all unit tests are fine, etc

Reply:
I'm not an authority on galley. But from my understanding when I moved pilot/envoy to golang/proto, is that galley is sensitive to gogo proto library use. It's not entirely clear to me why.
If you can exercise all galley paths handling protos, then we are good I guess.

Reply:
@ozevren do you think this is safe? Any ideas?

Reply:
@kyessenov @howardjohn @ozevren I would assume if the galley tests are passing that should be sufficient, no? Is there a particular path we're concerned about?

Reply:
Galley code uses (only) Gogo to perform certain operations (e.g. roll-in-out of Any, new'ing new instances, or cloning etc.) as that is what Istio has standardized on. There used to be dual support, but it has been fairly painful and permeated throughout the sources as the code in for such operations need to be duplicated and needed to track which particular proto was implemented using which flavor.

I don't think Gogo is "broken", per se, but it looks like removing k8s.io code somehow removes registrations for "glue-schema" needed K8s object-model specific types.

I think switching to (or staying with) either golang or gogo is fine from Galley's perspective, but mixing and matching is likely to cause problems (or alternatively, replumbing support throughout, so that the infrastructure can use the right Gogo v.s. Golang library, depending on the resource type). The above problem seems to be an issue appearing in the runtime, as it is doing an internal registry lookup to resolve. It is quite probable that some untested path can be silently broken.

Reply:
This is being fixed on K8s side, hopefully we can pick this up https://github.com/kubernetes/kubernetes/pull/83956/files

Reply:
I think we have 3 options:
1. Use golang here and gogo in other places. Seems like this may not be a great idea
2. Wait for the above PR to merge, and update our apimachinary dependency to kubernetes master. That means we have a mixer of kubernetes versions as dependencies, but it seems to work.
3. Drop this PR. I don't think we can do that since we need to keep the operator up to date.

Unless there are objections to #2 I will send a PR as soon as the kubernetes one merges